libraries/AP_HAL_ChibiOS/hwdef/scripts/dma_resolver.py

```diff
@@ -233,9 +233,9 @@ def generate_DMAMUX_map(peripheral_list, noshare_list, dma_exclude, stream_ofs):
     map1 = generate_DMAMUX_map_mask(dmamux1_peripherals, 0xFFFF, noshare_list, dma_exclude, stream_ofs)
     # there are 8 BDMA channels, but an issue has been found where if I2C4 and SPI6
     # use neighboring channels then we sometimes lose a BDMA completion interrupt. To
-    # avoid this we set the BDMA available mask to 0x33, which forces the channels not to be
+    # avoid this we set the BDMA available mask to 0x54, which forces the channels not to be
     # adjacent. This issue was found on a CUAV-X7, with H743 RevV.
-    map2 = generate_DMAMUX_map_mask(dmamux2_peripherals, 0x55, noshare_list, dma_exclude, stream_ofs)
+    map2 = generate_DMAMUX_map_mask(dmamux2_peripherals, 0x54, noshare_list, dma_exclude, stream_ofs)
```

Comment:
does this change previous mapping results?

Reply:
this one is complex (good EU dev call topic)
I did some more testing and found that both ADC3 and SPI6_RX can't use stream 0. On top of the existing known issues with BDMA I decided to just statically map the 3 peripherals that can use BDMA to fixed streams.
I also fixed SPI6 to stop trying to do BDMA locking, as we now can never have shared BDMA streams

Reply:
yeah, the change to using statically assigned channels make sense.